File: JAVA/Insertionsort.java
import java.util.Scanner;

public class Insertionsort{

public static void insertionsort(int arr[],int n){
    for(int i=0;i<n;i++){
        int j=i;
    while(j>0 && arr[j-1]>arr[j])// It checks the comparision of numbers until the ith position and it makes sorted for every iteration. 
    {
        int temp=arr[j-1];
        arr[j-1]=arr[j];
        arr[j]=temp;
        j--;
    }
    }
    System.out.println("After using InsertionSort");
    for(int i=0;i<n;i++){
        System.out.print(arr[i]+" ");
    }
    System.out.println();
}
    public static void main(String[] args){
        int arr[]={13,46,24,52,20,9};
        int n=arr.length;
       System.out.println("Before  using InsertionSort");
        for(int i=0;i<n;i++){
        System.out.print(arr[i]+" ");
        }
        System.out.println();
        insertionsort(arr,n);
    }
}

File: PYTHON/Insertion_sort.py
def insert(arr, i):
    key = arr[i]
    j = i - 1
    while j >= 0 and key < arr[j]:
        arr[j + 1] = arr[j]
        j -= 1
    arr[j + 1] = key

def insertionSort(arr):
    n = len(arr)
    for i in range(1, n):
        insert(arr, i)

# Example 1
arr1 = [10, 9, 8, 7, 6, 5, 4, 3, 2, 1]
insertionSort(arr1)
print("Sorted array (Example 1):", arr1)

# Example 2
arr2 = [4, 1, 3, 9, 7]
insertionSort(arr2)
print("Sorted array (Example 2):", arr2)
